code/modules/vore/eating/bellymodes_vr.dm

```diff
@@ -744,6 +769,214 @@
+						internal_contents -= P
+					if("Human")
+						var/obj/structure/closet/secure_closet/egg/human/J = new /obj/structure/closet/secure_closet/egg/human(O.loc)
+						P.forceMove(J)
+						J.name = "[defined_species] egg"
+						J.desc = "This egg has a very unique look to it."
+						internal_contents -= P
+					if("Slime")
+						var/obj/structure/closet/secure_closet/egg/slime/J = new /obj/structure/closet/secure_closet/egg/slime(O.loc)
+						P.forceMove(J)
+						J.name = "[defined_species] egg"
+						J.desc = "This egg has a very unique look to it."
+						internal_contents -= P
+					if("Egg")
+						var/obj/structure/closet/secure_closet/egg/J = new /obj/structure/closet/secure_closet/egg(O.loc)
+						P.forceMove(J)
+						J.name = "[defined_species] egg"
+						J.desc = "This egg has a very unique look to it."
+						internal_contents -= P
+					if("Xenochimera")
+						var/obj/structure/closet/secure_closet/egg/scree/J = new /obj/structure/closet/secure_closet/egg/scree(O.loc)
+						P.forceMove(J)
+						J.name = "[defined_species] egg"
+						J.desc = "This egg has a very unique look to it."
+						internal_contents -= P
+					if("Xenomorph")
+						var/obj/structure/closet/secure_closet/egg/xenomorph/J = new /obj/structure/closet/secure_closet/egg/xenomorph(O.loc)
+						P.forceMove(J)
+						J.name = "[defined_species] egg"
+						J.desc = "This egg has a very unique look to it."
+						internal_contents -= P
+					else
+						var/obj/structure/closet/secure_closet/egg/J = new /obj/structure/closet/secure_closet/egg(O.loc)
+						P.forceMove(J)
+						J.name = "[defined_species] egg"
+						J.desc = "This egg has a very unique look to it."
+						internal_contents -= P
+			else
+				P.r_hair 			= O.r_hair
+				P.r_facial 			= O.r_facial
+				P.g_hair 			= O.g_hair
+				P.g_facial 			= O.g_facial
+				P.b_hair 			= O.b_hair
+				P.b_facial 			= O.b_facial
+				P.r_skin 			= O.r_skin
+				P.g_skin 			= O.g_skin
+				P.b_skin 			= O.b_skin
+				P.tail_style 		= O.tail_style
+				P.r_tail 			= O.r_tail
+				P.g_tail 			= O.g_tail
+				P.b_tail 			= O.b_tail
+				P.ear_style 		= O.ear_style
+				P.h_style 			= O.h_style
+				P.species 			= O.species
+				P.custom_species 	= O.custom_species
+				P.r_eyes 			= O.r_eyes
+				P.g_eyes 			= O.g_eyes
+				P.b_eyes 			= O.b_eyes
+				for(var/obj/item/organ/I in P.internal_organs) //This prevents organ rejection
+					I.species = O.species
+				for(var/obj/item/organ/external/chest/A in O.organs)
+					for(var/obj/item/organ/external/Z in P.organs) //This makes their limb sprites look correct.
+						Z.species = O.species
+						Z.h_col = A.h_col
+						Z.s_col = A.s_col
+				P << "<span class='notice'>You lose sensation of your body, feeling only the warmth around you as are you as you're encased in an egg. </span>"
+				owner << "<span class='notice'>You shift as you make dramatic changes to your captive's body as you encase them in an egg.</span>"
+				P.fixblood()
```

Comment:
This gives you all your blood back...? So eating someone, doing TF for 1 tick to something if they have 10% blood, and then defibbing works fine? :P Seems a little odd.